core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -1094,6 +1095,14 @@ public Node getNode(String name) {
 	}
 
 	// Events
+	public EventList getEventList() {
+		return new EventList(events().list().getItems());
```

Comment:
So have you already looked ifevents cant be filtered? E.g by `fieldSelector` or `watch`?

[1] https://docs.openshift.com/container-platform/3.11/rest_api/api/v1.Event.html#Get-api-v1-namespaces-namespace-events

Reply:
**fieldSelector**
This enables me to restrict the list by object. I can not use it multipe times - https://access.redhat.com/solutions/2126531

For example:
```
29m       Normal    Killing             pod/s2i-with-cli-env-1-j2j76               Stopping container s2i-with-cli-env
49m       Normal    SuccessfulCreate    replicationcontroller/s2i-with-cli-env-1   Created pod: s2i-with-cli-env-1-drbzn
``` 
I need to do 2 request targeting both names. I can not use regex or anything. 
Even If I ended up only at 2 request, the overhead just too big. Retrieving 57 events takes 1.6second and retrieving 5 events 1.1 

**watch**
This would bring another complexity to XTF. It would needed to run in parallel (it is a stream of events). The thread would need to be terminated properly in all cases. I do not think it is worth the effort.

I have implemented ExponentialBackoff to reduce the overhead. This means events are retrieved in 1st second, 2nd, 4th, 8th, 16th, 32th, 32th, 32th,....moreover the waiting is non-blocking.

Reply:
Update:
It seems we can put more fieldSelector into query (I just tried & - my bad). However per internal discussion and experiments It seems It is AND and we still can't select all events.